projects/python-client/src/datapane/client/api/user.py

```diff
@@ -113,6 +116,76 @@ def _run_script(script: str):
     runpy.run_path(str(script_path), run_name="__datapane__")
 
 
+def _run_template(template_path: str):
+    """Run the template script"""
+    # Store the cwd before we change it
+    old_cwd = os.getcwd()
+    # Change cwd to support relative paths in template script
+    os.chdir(template_path)
+
+    try:
+        runpy.run_path(
+            "template.py",
+            run_name="__datapane__",
+        )
+    # Notify the user of missing packages that are required by the template
+    except ModuleNotFoundError as e:
+        raise DPError(f"Please install the following packages to run this template\n{e.name}") from e
```

Comment:
do you think we should use something like `pipreqs` here to auto create the requirements.txt for any missing packages? I wonder if we can install them as well?

Reply:
I'm going to check out `pipreqs` for generating requirements.txt for convenience alongside the shell output, and let users manage their own environments based on that.

Reply:
@mands `pipreqs` doesn't currently have a Python API (https://github.com/bndr/pipreqs/issues/108). If we use `pipreqs`, what's your opinion on using it via subprocess?

Reply:
Following the updated 80/20 pitch - we won't be looking at pipreqs in this PR.